crates/credentials/src/pex/v2/presentation_definition.rs

```diff
@@ -0,0 +1,230 @@
+use jsonschema::{Draft, JSONSchema};
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use serde_with::skip_serializing_none;
+use std::collections::HashMap;
+
+/// Presentation Exchange
+///
+/// Presentation Exchange specification codifies a Presentation Definition data format Verifiers
+/// can use to articulate proof requirements, and a Presentation Submission data format Holders can
+/// use to describe proofs submitted in accordance with them.
+///
+/// See [Presentation Definition](https://identity.foundation/presentation-exchange/#presentation-definition)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct PresentationDefinition {
+    pub id: String,
+    pub name: Option<String>,
+    pub purpose: Option<String>,
+    pub format: Option<Format>,
+    pub submission_requirements: Option<Vec<SubmissionRequirement>>,
+    pub input_descriptors: Vec<InputDescriptor>,
+    pub frame: Option<HashMap<String, JsonValue>>,
+}
+
+/// Represents an input descriptor in a presentation definition.
+///
+/// See [Input Descriptor](https://identity.foundation/presentation-exchange/#input-descriptor-object)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct InputDescriptor {
+    pub id: String,
+    pub name: Option<String>,
+    pub purpose: Option<String>,
+    pub format: Option<Format>,
+    pub constraints: Constraints,
+}
+
+/// Represents constraints for an input descriptor.
+///
+/// See 'constraints object' defined in
+/// [Input Descriptor](https://identity.foundation/presentation-exchange/#input-descriptor-object)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct Constraints {
+    pub fields: Option<Vec<Field>>,
+    pub limit_disclosure: Option<ConformantConsumerDisclosure>,
+}
+
+/// Represents a field in a presentation input descriptor.
+///
+/// See 'fields object' as defined in
+/// [Input Descriptor](https://identity.foundation/presentation-exchange/#input-descriptor-object)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct Field {
+    pub id: Option<String>,
+    pub path: Vec<String>,
+    pub purpose: Option<String>,
+    pub filter: Option<JsonValue>,
+    pub predicate: Option<Optionality>,
+    pub name: Option<String>,
+    pub optional: Option<bool>,
+}
+
+impl Field {
+    pub fn filter_schema(&self) -> Option<JSONSchema> {
+        self.filter
+            .as_ref()
+            .map(|json| {
+                JSONSchema::options()
+                    .with_draft(Draft::Draft7)
```

Comment:
What is the concept of a "draft" here?

Reply:
it's a revision for json schema itself...unfortunately they're still using quite an old draft